core/src/main/java/org/apache/iceberg/deletes/Deletes.java

```diff
@@ -135,6 +142,17 @@ public static StructLikeSet toEqualitySet(
     }
   }
 
+  static ExecutorService getDeletePosThreadPool() {
+    return (SystemProperties.getBoolean(SystemProperties.DELETE_POS_FILES_THREADS_ENABLED, false))
+        ? ThreadPools.newWorkerPool(SystemProperties.DELETE_POS_FILES_THREADS_ENABLED)
+        : null;
```

Comment:
Should this be in ThreadPools like the worker pool? I think setting pool size there like we do for the worker pool makes sense.

Reply:
Addressed this in recent PR. Moved it to ThreadPools.newDeleteWorkerPool